src/@types/EventHandler.d.ts

```diff
@@ -0,0 +1,25 @@
+// IDK why this line is necessary, but if I remove it, I can't use EventHandler anywhere
+import type { Awaitable } from 'discord.js';
+
+declare global {
+	/**
+	 * Defines an event handler.
+	 */
+	interface EventHandler {
```

Comment:
I think we can make this interface loads safer by making it generic over the keys of the `ClientEvents` interface from discord.js. We can then augment the `name` and `execute` properties according to the event name. This way, event handlers always know exactly what data to expect from the API

Reply:
Could we save that for a future PR? I'm still learning TypeScript and it's been mostly trial-and-error so far. That's probably better left to someone who knows what their doing.

Reply:
I'll get right on it once this is merged

Reply:
I suggest making an issue for it before we forget.